src/Fixer/ClassNotation/OrderedClassElementsFixer.php

```diff
@@ -0,0 +1,390 @@
+     *
+     * @return array[]
+     */
+    private function getElements(Tokens $tokens, $startIndex)
+    {
+        static $elementTokenKinds = array(CT_USE_TRAIT, T_CONST, T_VARIABLE, T_FUNCTION);
+
+        ++$startIndex;
+        $elements = array();
+
+        while (true) {
+            $element = array(
+                'start' => $startIndex,
+                'visibility' => 'public',
+                'static' => false,
+            );
+
+            for ($i = $startIndex; ; ++$i) {
+                $token = $tokens[$i];
+
+                // class end
+                if ($token->equals('}')) {
+                    return $elements;
+                }
+
+                if ($token->isGivenKind(T_STATIC)) {
+                    $element['static'] = true;
+                    continue;
+                }
+
+                if ($token->isGivenKind(array(T_PROTECTED, T_PRIVATE))) {
+                    $element['visibility'] = strtolower($token->getContent());
+                    continue;
+                }
+
+                if (!$token->isGivenKind($elementTokenKinds)) {
+                    continue;
+                }
+
+                $type = $this->detectElementType($tokens, $i);
+                if (is_array($type)) {
+                    $element['type'] = $type[0];
+                    $element['name'] = $type[1];
+                } else {
+                    $element['type'] = $type;
+                }
+
+                $element['end'] = $this->findElementEnd($tokens, $i);
+                break;
+            }
+
+            $elements[] = $element;
+            $startIndex = $element['end'] + 1;
+        }
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $index
+     *
+     * @return string|array type or array of type and name
+     */
+    private function detectElementType(Tokens $tokens, $index)
+    {
+        $token = $tokens[$index];
+
+        if ($token->isGivenKind(CT_USE_TRAIT)) {
+            return 'use_trait';
+        }
+
+        if ($token->isGivenKind(T_CONST)) {
+            return 'constant';
+        }
+
+        if ($token->isGivenKind(T_VARIABLE)) {
+            return 'property';
+        }
+
+        $nameToken = $tokens[$tokens->getNextMeaningfulToken($index)];
+
+        if ($nameToken->equals(array(T_STRING, '__construct'), false)) {
+            return 'construct';
+        }
+
+        if ($nameToken->equals(array(T_STRING, '__destruct'), false)) {
+            return 'destruct';
+        }
+
+        if (
+            $nameToken->equalsAny(array(
+                array(T_STRING, 'setUpBeforeClass'),
+                array(T_STRING, 'tearDownAfterClass'),
+                array(T_STRING, 'setUp'),
+                array(T_STRING, 'tearDown'),
+            ), false)
+        ) {
+            return array('phpunit', strtolower($nameToken->getContent()));
+        }
+
+        if ('__' === substr($nameToken->getContent(), 0, 2)) {
+            return 'magic';
```

Comment:
Should I use a static array containing all magic method names, or just checking for names beginning with `__`?

http://php.net/manual/en/language.oop5.magic.php

> Caution
PHP reserves all function names starting with __ as magical. It is recommended that you do not use function names with __ in PHP unless you want some documented magic functionality.